### PR TITLE
(waterfox) Classic and Current support

### PR DESCRIPTION
**Pull request by RedBaron2**
@gep13 @pascalberger @majkinetor @AdmiringWorm 
## Description

This will make the waterfox package a stream package to support Classic and Current
The Classic readme might need a hyperlink to Current

## Motivation and Context

The issue #1440 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Migrated package (a package has been migrated from another repository)

## Checklist:

- [x] My code follows the code style of this repository.
- [ ] My change requires a change to documentation (this usually means the notes in the description of a package).
- [ ] I have updated the documentation accordingly (this usually means the notes in the description of a package).
- [x] All files are up to date with the latest [Contributing Guidelines](https://github.com/chocolatey/chocolatey-coreteampackages/blob/master/CONTRIBUTING.md)
- [x] The added/modified package passed install/uninstall in the chocolatey test environment.
- [x] The changes only affect a single package (not including meta package).

## Original Location
- [Original Repository](add_link_to_original_repository_location)
- [Open Issues](link_to_the_generic_location_of_open_issues) *Add the different issues underneath, and tick those that are fixed in this PR*
  - [x] Issue 1 #1440 
  - [ ] Issue 2 Link
- [ ] *Include the link to the opened PR that removes the package from the original location*
- [ ] The [migration guidelines](https://github.com/chocolatey/chocolatey-coreteampackages/wiki/Package-migration-process) have been followed

### CODE REVIEW

**Comment by RedBaron2** [comment]
Just creating a empty file named `waterfox.json` cause a fault, but making it have `{ }` fixed the error from yesterday.
Ready for your reviews

File: automatic/waterfox/Readme.classic.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# <img src="https://cdn.jsdelivr.net/gh/chocolatey-community/chocolatey-coreteampackages@2171e76d9468526e4e792d20ac6b68e1a8fdc93a/icons/waterfox.png" width="48" height="48"/> [waterfox](https://chocolatey.org/packages/waterfox)
+
+## How and why Waterfox is here...Waterfox Classic
```
Comment (AdmiringWorm) [inline automatic/waterfox/Readme.classic.md:3]: Do we need a section header for this?
Maybe just making the paragraph bold would be better instead?

Not completely sure, it just feels off to me.

File: automatic/waterfox/Readme.current.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# <img src="https://cdn.jsdelivr.net/gh/chocolatey-community/chocolatey-coreteampackages@2171e76d9468526e4e792d20ac6b68e1a8fdc93a/icons/waterfox.png" width="48" height="48"/> [waterfox](https://chocolatey.org/packages/waterfox)
+
+## How and why Waterfox is here...
```
Comment (AdmiringWorm) [inline automatic/waterfox/Readme.current.md:3]: Same as the classic readme, having this as a section header seems off to me.

File: automatic/waterfox/update.ps1 [hunk 3]
```diff
@@ -1,9 +1,15 @@
-﻿import-module au
-
+import-module au
```
Comment (AdmiringWorm) [inline automatic/waterfox/update.ps1:1]: Looks like you have removed the UTF-8 BOM encoding on this file.
For compatibility reasons, it is best not to remove that on Powershell files.

File: automatic/waterfox/update.ps1 [hunk 4]
```diff
@@ -25,23 +31,40 @@ function global:au_SearchReplace {
       "(?i)(^\s*softwareName\s*=\s*)'.*'" = "`${1}'$softwareName'"
       "(?i)(^\s*packageName\s*=\s*)'.*'"  = "`${1}'$($Latest.PackageName)'"
     }
+    ".\waterfox.nuspec"  = @{
+      "(?i)(^\s*\<id\>).*(\<\/id\>)" = "`${1}$($Latest.PackageName)`${2}"
```
Comment (AdmiringWorm) [inline automatic/waterfox/update.ps1:35]: Please remove this regex. The package id will already be set to the `PackageName` property used in the stream.

File: automatic/waterfox/update.ps1 [hunk 5]
```diff
@@ -25,23 +31,40 @@ function global:au_SearchReplace {
...
     $re = 'Setup\.exe$' # If we didn't get a url with the previous regex, we use a much simpler way
     $url = $download_page.links | ? href -match $re | select -First 1 -expand href
   }
 
   $version  = $url -split '%20| ' | select -Last 1 -Skip 1
-
+  if ($build -eq 'Classic') { $build = 'Classic'; $dash = '-' } else { $build=$dash = '' }
   # We need to replace the space in the url, otherwise we'll get an invalid url error.
-  return @{ URL64 = ($url -replace ' ','%20'); Version = $version }
+  return @{ PackageName = "Waterfox$dash$build" ; Title = "Waterfox $build" ; URL64 = ($url -replace ' ','%20'); Version = $version }
```
Comment (AdmiringWorm) [inline automatic/waterfox/update.ps1:57]: You will need to do some more special handling here.
The package name for the **current** edition of Waterfox should be `Waterfox` (yes, with uppercase W), but for the **classic** edition, it should be `waterfox-classic` all in lowercase characters.

File: automatic/waterfox/update.ps1 (hunk 3 above)
Comment (RedBaron2) [reply automatic/waterfox/update.ps1:1]: done by mistake

File: automatic/waterfox/update.ps1 (hunk 4 above)
Comment (RedBaron2) [reply automatic/waterfox/update.ps1]: done

File: automatic/waterfox/update.ps1 (hunk 5 above)
Comment (RedBaron2) [reply automatic/waterfox/update.ps1]: fixed

**Comment by mstick99** [comment]
Waterfox Classic new version 2020.4 is available.

**Comment by mstick99** [comment]
Waterfox Classic new version 2020.4 available.

**Comment by AdmiringWorm** [comment]
@mstick99 we can not update the package until the first version is approved.
After that has happened, the package will be updated when there is a new version together with every other package in the repository (every 6th hour).

**Pull request merged**